File: src/AI.Forged.TourOps.Domain/Entities/Supplier.cs
namespace AI.Forged.TourOps.Domain.Entities;

public class Supplier
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public DateTime CreatedAt { get; set; }

    public ICollection<Product> Products { get; set; } = new List<Product>();
}

File: ui/src/types/supplier.ts
export interface SupplierListItem {
  id: string;
  name: string;
  email?: string | null;
  phone?: string | null;
}

export interface Supplier {
  id: string;
  name: string;
  email?: string | null;
  phone?: string | null;
  createdAt: string;
}

export interface SupplierRequest {
  name: string;
  email?: string | null;
  phone?: string | null;
}
